packages/clay-badge/src/ClayBadge.soy

```diff
@@ -0,0 +1,69 @@
+{namespace ClayBadge}
+
+/**
+ * This renders a badge.
+ */
+{template .render}
+	{@param? closeable: bool}
+	{@param? href: string}
+	{@param? id: string}
+	{@param? label: html|string}
+	{@param? size: string}
+	{@param? spritemap: string}
+	{@param? style: string}
+
+	{let $classes kind="text"}
+		badge badge-pill
+		{sp}badge-{$style ?: 'default'}
+
+		{if $size}
+			{sp}badge-{$size}
+		{/if}
+	{/let}
+
+	{let $attributes kind="attributes"}
+		class="{$classes}"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+	{/let}
+
+	{if $href}
+		{call ClayLink.render}
+			{param elementClasses: $classes /}
+			{param href: $href /}
+			{param id: $id /}
+			{param label: $label /}
+		{/call}
+	{else}
+		<span {$attributes}>
+			{call .content}
+				{param closeable: $closeable /}
+				{param label: $label /}
+				{param spritemap: $spritemap /}
+			{/call}
+		</span>
+	{/if}
+
+{/template}
+
+/**
+ * Renders icon content
+ */
+{template .content}
+	{@param? closeable: bool}
+	{@param? label: html|string}
+	{@param? spritemap: string}
+
+	{$label ? $label : ''}
```

Comment:
I don't see where this `label` param is being used. Do we need it?

Reply:
Yes, when we do not pass `href` and we want to render only the `label` with icon or without.

Reply:
Ah .Got it